oma-contents/src/searcher.rs

```diff
@@ -87,6 +87,36 @@ impl Mode {
     }
 }
 
+/// Automatically selects a search method based on whether or not the `rg` binary is used.
+///
+/// If the `rg` binary exists, the `rg` search is called, and if it doesn't, the fallback
+/// is to a search method that doesn't depend on a binary.
+///
+/// # Arguments
+///
+/// * `dir` - A reference to a `Path` that specifies the directory to search in.
+/// * `mode` - A `Mode` enum value that specifies the type of search operation to perform.
+/// * `query` - A string slice that contains the search query.
+/// * `cb` - A mutable callback function that takes a tuple of two strings (the matched line and the matched part) as its argument.
+///
+/// # Returns
+///
+/// Returns a `Result` which is `Ok(())` if the search is successful, or an `OmaContentsError` if an error occurs.
```

Comment:
Returns a `Result` - `Ok(())` if the search is successful, `OmaContentsError` if not.